### DISCUSSION TITLE
unresolved dependency: sbt-plugin

### DISCUSSION THREAD

**Discussion by azadbolour** [playframework/playframework#11750] [General]
I am getting the following unresolved dependency in building my Scala Play
project:

[warn] 	Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn] 		com.typesafe.play:sbt-plugin:2.8.7 (sbtVersion=1.0, scalaVersion=2.12)
[warn]
[warn] 	Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading com.typesafe.play:sbt-plugin;sbtVersion=1.0;scalaVersion=2.12:2.8.7
[error]   Not found
[error]   Not found
[error]   not found: https://repo1.maven.org/maven2/com/typesafe/play/sbt-plugin_2.12_1.0/2.8.7/sbt-plugin-2.8.7.pom
[error]   not found: /Users/azadbolour/.ivy2/localcom.typesafe.play/sbt-plugin/scala_2.12/sbt_1.0/2.8.7/ivys/ivy.xml
[error]   checksum format error: /Users/azadbolour/Library/Caches/Coursier/v1/https/repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.play/sbt-plugin/scala_2.12/sbt_1.0/2.8.7/ivys/ivy.xml.sha1
[error]   checksum format error: /Users/azadbolour/Library/Caches/Coursier/v1/https/repo.typesafe.com/typesafe/ivy-releases/com.typesafe.play/sbt-plugin/scala_2.12/sbt_1.0/2.8.7/ivys/ivy.xml.sha1

My setup includes:

build.sbt with:
  enablePlugins(PlayScala)
  scalaVersion := "2.12.12"

build.properties:
  sbt.version=1.8.2

plugins.sbt
  logLevel := Level.Warn
  resolvers += Resolver.sbtPluginRepo("releases")
  addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.7")

The same issue remains for version 2.8.0 of Play. And for version 1.3.13 of sbt.

How should I configure my project to access the needed play sbt plugin?

Many thanks.

Azad

**Comment by joel-apollo** [comment]
This is caused by https://github.com/sbt/sbt/issues/7202 and should now be fixed

**Reply by mkurz** [reply]
Should be back online now however.

**Comment by azadbolour** [comment]
Thanks for your prompt response.

My understanding is that the plugin should now be available under:

https://repo1.maven.org/maven2/com/typesafe/play/sbt-plugin_2.12_1.0/

However, the lowest version of the plugin there is 2.8.8. If I use that version, I get the following incompatibility:
```
[error] java.lang.RuntimeException: found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.scala-lang.modules:scala-xml_2.12:2.1.0 (early-semver) is selected over {1.2.0, 1.1.1}
[error] 	    +- org.scala-lang:scala-compiler:2.12.17              (depends on 2.1.0)
[error] 	    +- com.typesafe.sbt:sbt-native-packager:1.5.2 (scalaVersion=2.12, sbtVersion=1.0) (depends on 1.1.1)
[error] 	    +- com.typesafe.play:twirl-api_2.12:1.5.1             (depends on 1.2.0)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
```
And I am not sure what to do about that.

If the lower versions are no longer supported, can you advise on working around the above incompatibilities?

If lower versions are still supported, I was hoping to use version 2.8.7, or in case of issues, version 2.8.0. Could they be added to the repo?

Many thanks.

Azad

**Reply by mkurz** [reply]
Do you use sbt 1.8.2 in your `project/build.properties`? If so try downgrading to 1.7.2 in that file and try again.

**Reply by joelhaasnoot** [reply]
There's also a note about how to fix this in the release guide for 2.8.19 https://github.com/playframework/playframework/releases/tag/2.8.19

**Reply by azadbolour** [reply]
Thanks for the tips. Using sbt 1.7.2 with play version 2.8.8 I was able to get my build to complete successfully.

Thanks again for the prompt responses. Much appreciated.

Azad
